src/libstd/sys/unix/ext.rs

```diff
@@ -220,6 +224,27 @@ pub mod fs {
             self.as_inner_mut().mode(mode); self
         }
     }
+
+    /// Creates a new symbolic link on the filesystem.
+    ///
+    /// The `dst` path will be a symbolic link pointing to the `src` path.
```

Comment:
Could you also add some notes about how windows requires a special permission to create symlinks? Otherwise this all looks great to me!